alias.go

```diff
@@ -49,21 +57,25 @@ func (r *ResponseModifier) WriteMsg(res *dns.Msg) error {
 		cname string
 		ttl   uint32
 	)
-	for i, rr := range res.Answer {
-		if rr.Header().Rrtype == dns.TypeCNAME && rr.Header().Name == zone {
+	cname = zone
+	ttl = math.MaxUint32
+	for i := 0; i < len(res.Answer); {
+		rr := res.Answer[i]
+		if rr.Header().Rrtype == dns.TypeCNAME && rr.Header().Name == cname {
 			cname = rr.(*dns.CNAME).Target
-			ttl = rr.(*dns.CNAME).Header().Ttl
+			ttl = min(ttl, rr.(*dns.CNAME).Header().Ttl)
 			// Remove the CNAME record
 			res.Answer = append(res.Answer[:i], res.Answer[i+1:]...)
-			break
+			continue
```

Comment:
Where feasible I prefer to PR upstream. Guessing we'd have a hard time getting this change merged. Should we TP to make this logic change a config option so we could merge upstream?